core/chainio/builder.go

```diff
@@ -0,0 +1,214 @@
+	"errors"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/avsregistry"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/eth"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/wallet"
+	"github.com/Layr-Labs/eigensdk-go/chainio/txmgr"
+	chainioutils "github.com/Layr-Labs/eigensdk-go/chainio/utils"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/signerv2"
+	"github.com/Layr-Labs/eigensdk-go/types"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+type Clients struct {
+	AvsRegistryChainReader     *avsregistry.AvsRegistryChainReader
+	AvsRegistryChainSubscriber *avsregistry.AvsRegistryChainSubscriber
+	AvsRegistryChainWriter     *avsregistry.AvsRegistryChainWriter
+	ElChainReader              *elcontracts.ELChainReader
+	ElChainWriter              *elcontracts.ELChainWriter
+}
+
+func BuildAll(
+	avsName string,
+	registryCoordinatorAddr string,
+	operatorStateRetrieverAddr string,
+	ethHttpClient eth.Client,
+	ethWsClient eth.Client,
+	ecdsaPrivateKey *ecdsa.PrivateKey,
+	logger logging.Logger,
+) (*Clients, error) {
+	rpcCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	chainid, err := ethHttpClient.ChainID(rpcCtx)
+	if err != nil {
+		return nil, types.WrapError(errors.New("cannot get chain id"), err)
+	}
+
+	signerV2, addr, err := signerv2.SignerFromConfig(signerv2.Config{PrivateKey: ecdsaPrivateKey}, chainid)
+	if err != nil {
+		return nil, types.WrapError(errors.New("cannot create signer from config"), err)
+	}
+
+	pkWallet, err := wallet.NewPrivateKeyWallet(ethHttpClient, signerV2, addr, logger)
+	if err != nil {
+		return nil, types.WrapError(errors.New("failed to create transaction sender"), err)
+	}
+
+	txMgr := txmgr.NewSimpleTxManager(pkWallet, ethHttpClient, logger, addr)
+
+	// creating EL clients: Reader, Writer and Subscriber
+	elChainReader, elChainWriter, err := buildElClients(
+		registryCoordinatorAddr,
+		operatorStateRetrieverAddr,
+		ethHttpClient,
+		txMgr,
+		logger,
+	)
+	if err != nil {
+		return nil, types.WrapError(errors.New("failed to create EL Reader, Writer and Subscriber"), err)
+	}
+
+	// creating AVS clients: Reader and Writer
+	avsRegistryChainReader, avsRegistryChainSubscriber, avsRegistryChainWriter, err := buildAvsClients(
+		registryCoordinatorAddr,
+		operatorStateRetrieverAddr,
+		elChainReader,
+		ethHttpClient,
+		ethWsClient,
+		txMgr,
+		logger,
+	)
+	if err != nil {
+		return nil, types.WrapError(errors.New("failed to create AVS Registry Reader and Writer"), err)
+	}
+
+	return &Clients{
+		ElChainReader:              elChainReader,
+		ElChainWriter:              elChainWriter,
+		AvsRegistryChainReader:     avsRegistryChainReader,
+		AvsRegistryChainSubscriber: avsRegistryChainSubscriber,
+		AvsRegistryChainWriter:     avsRegistryChainWriter,
+	}, nil
+
+}
+
+func buildElClients(
```

Comment:
It could be argued that we should expose these functions instead of `BuildAll` to allow for more granularity. For now we'll keep the currently used API